src/lib/services/project-service.ts

```diff
@@ -671,6 +671,19 @@ export default class ProjectService {
             projectId,
             userId,
         );
+
+        const ownerRole = await this.accessService.getRoleByName(
+            RoleName.OWNER,
+        );
+
+        if (
```

Comment:
It's early and I don't have a lot of coffee in my blood yet but gosh this if statement is hard to read. It's also duplicated. 

Can we pull this out into a function and give some of the contents some names? ChatGPT suggests something like this:

```
const hasExistingOwnerRole = existingRoles.includes(ownerRole.id);
const isRemovingOwnerRole = !roles.includes(ownerRole.id);

if (hasExistingOwnerRole && isRemovingOwnerRole) {
    await this.validateAtLeastOneOwner(projectId, ownerRole);
}

```

Reply:
What I don't like about adding constants is that despite the short-circuit `&&` we have to execute both finds, but I know what to do

Reply:
Is that actually a concern? `find` executes in about a millisecond on 100K elements (how many roles do we have?). Piping stuff back from the database is going to dwarf this completely. 

If it offends you, go for it, but I don't think it's necessary to trade off readability for performance

Reply:
Yeah, when I was making the change I had the same thought and I'm leaning towards readability on this one ;)